src/custom/hooks/useGetGpApiStatus.ts

```diff
@@ -0,0 +1,39 @@
+import ms from 'ms.macro'
+import { useState, useEffect } from 'react'
+
+export type GpQuoteStatus = 'COWSWAP' | 'LEGACY'
+// TODO: use actual API call
+export async function checkGpQuoteApiStatus(): Promise<GpQuoteStatus> {
+  return new Promise((accept) => setTimeout(() => accept('LEGACY'), 500))
+}
+const GP_QUOTE_STATUS_INTERVAL_TIME = ms`2 hours`
+
+export default function useCheckGpQuoteStatus(defaultApiToUse: GpQuoteStatus): GpQuoteStatus {
+  const [gpQuoteApiStatus, setGpQuoteApiStatus] = useState<GpQuoteStatus>(defaultApiToUse)
+
+  useEffect(() => {
+    console.debug('[useGetQuoteCallback::GP API Status]::', gpQuoteApiStatus)
+
+    const checkStatus = () => {
+      checkGpQuoteApiStatus()
+        .then(setGpQuoteApiStatus)
+        .catch((err: Error) => {
+          console.error('[useGetQuoteCallback::useEffect] Error getting GP quote status::', err)
+          // Fallback to LEGACY
+          setGpQuoteApiStatus('LEGACY')
+        })
+    }
+
+    // Create initial call on mount
+    checkStatus()
+
+    // set interval for GP_QUOTE_STATUS_INTERVAL_TIME (2 hours)
+    const intervalId = setInterval(() => {
+      checkStatus()
+    }, GP_QUOTE_STATUS_INTERVAL_TIME)
+
+    return () => clearInterval(intervalId)
+  }, [gpQuoteApiStatus])
```

Comment:
This hook will schedule a checkStatus on every time you use it. 

I think our project could use something like:
- https://react-query.tanstack.com/
- https://swr.vercel.app

These libraries solve exactly this problems in an elegant way. i also talked with @alfetopito about this. I think this data logics are something that we reinvent the wheel every time (fetching, caching, invalidating, coordinating multiple quering of the same endpoint, loding flags, error handling, etc)

I think we should choose one of the two.

I reviewed SRW and its great, and very performant! on most comparisons they show it as a superior approach than React Query in terms of performance and production readyness. Although, React Query has some more features (and bundle size).

Reply:
ok we can add that library later... definitely not for this PR

Reply:
i can consider making it a top level updater so it's only called once

Reply:
yes, fair to not mix the new library with this PR. Although not too bad to use one new feature to test the new library out. Not necessarily in this PR